### DISCUSSION TITLE
Quartz Misfire After Service Restart

### DISCUSSION THREAD

**Discussion by aarondglover** [quartznet/quartznet#1350] [Q&A]
Assume the following;

A persistent ADOJob Store.

A job which is scheduled with a cron statement to fire every 1 minute.

Misfire Threshold set to 10 seconds

Service is stopped for 10 minutes.

On restart is there anyway for those 10 "missed" trigger instances to execute on startup?

I can seemingly only get ONE of the misfires to execute on startup.

Any guidance is greatly appreciated

Thanks
Aaron

**Comment by lahma** [comment]
Such backlog processing not currently supported. Only SimpleTrigger can go into this mode if you specifically want it to do so, but that would still be an avalanche of triggers to handle. See [MisfireInstruction.cs](https://github.com/quartznet/quartznet/blob/main/src/Quartz/MisfireInstruction.cs) for more details.

One could easily configure a trigger to fire every 5 seconds and then shutdown the server for the night and in the morning there would be quite the fireworks handling the backlog on top of the current ones (of course depends on what the jobs actually do).

Without knowing the use case behind, I would try to model business critical tasks to have separate entity in database that will hold information whether it actually has been processed. I can see a relation with fire-once single trigger and a record in database but for cron it's harder for me to see which are affected (again not knowing what is happening in your use case). Compensation logic should kick in if there's something missing and re-schedule or process things still needing attention.

Hope this helps.
